python/lsst/dax/ppdb/sql/_chunk_exporter.py

```
@@ -0,0 +1,187 @@
+# This file is part of dax_ppdb
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import os
+import pyarrow as pa
+import pyarrow.parquet as pq
+from io import BytesIO
+from datetime import datetime, timezone
+
+from ._ppdb_sql import PpdbSql
+from lsst.dax.apdb import ApdbTableData, ReplicaChunk
+
+try:
+    from google.cloud import storage
+except ImportError:
+    storage = None
```

The code does not do anything useful is `storage` is `None`, should it just crash here?

---

Maybe. I was following your philosophy of having classes be loadable without "optional" dependencies, but if the entire module should depend on the GCS package being installed, I can just have it fail here (would make the rest of the module simpler).

---

Well, if `google.cloud` is optional (missing from environment) then you can probably keep this, otherwise `ppdb-replication` CLI will crash on import of this module. Still, there is no point in trying to instantiate `ChunkExporter` when `storage` is `None`, you should raise an exception in its constructor in that case.